src/xlf/hooks/index.ts

```diff
@@ -1,10 +1,27 @@
-import {CustomRendererLifeCycle} from '@diplodoc/markdown-it-custom-renderer';
+import {CustomRendererHook, CustomRendererHooks, CustomRendererLifeCycle} from '@diplodoc/markdown-it-custom-renderer';
 
 import template, {TemplateParameters} from './template';
 import meta, {MetaParameters} from './meta';
 import {includes} from './diplodoc/includes';
 
-export type HooksParameters = TemplateParameters & MetaParameters;
+type CustomHooks = {customHooks?: CustomRendererHooks};
+
+export type HooksParameters = TemplateParameters & MetaParameters & CustomHooks;
+
+const getCustomHooks = (lifecycle: 'BeforeRender' | 'AfterRender', customHooks?: CustomRendererHooks): CustomRendererHook[] => {
```

Comment:
lifecycle type should be: `lifecycle: CustomRendererLifeCycle`

also there is more concise way to express this logic:

```
if (!custom) {
    return []
}
return Array.isArray(custom) ? custom : [custom]
```

Reply:
actually i think you can even do 
```
function get...(custom = []) {
    return Array.isArray(custom) ? custom : [custom]
}
```
and skip if check...

Reply:
> actually i think you can even do
> 
> ```
> function get...(custom = []) {
>     return Array.isArray(custom) ? custom : [custom]
> }
> ```
> 
> and skip if check...

This can produce `[undefined]`

Reply:
not sure i understand

we are providing the default value to avoid that.

[ts-playground](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABFAFAZzQLkWqAnGMAcwG0BdRAHx30KMQF5FyBKRAbwChEfEIFciPI0QBBPHgCGATwB0MNOKnT0aNgH4caRNhIYyAbm69+YNHAA2AU1kW4RFHhZGAvp06oWHlK04B6P2QUMBALCy9UcAATK2BCKyiInwBySWSyFiA)